src/main/java/nextstep/subway/domain/SectionLineUp.java

```diff
@@ -0,0 +1,132 @@
+
+    @JoinColumn(name = "line_id")
+    @OneToMany(orphanRemoval = true, cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<Section> sectionList = new ArrayList<>();
+
+    protected SectionLineUp() {
+    }
+
+    public List<Section> getSectionList() {
+        return sectionList;
+    }
+
+    public void init(Line line, Section section) {
+        if (!sectionList.isEmpty()) {
+            throw new IllegalStateException("새로 생성된 노선이 아닙니다");
+        }
+        sectionList.add(section);
+        section.registerLine(line);
+    }
+
+    public void add(Line line, Section section) {
+        if (!sectionList.isEmpty()) {
+            validUnknownStation(section);
+        }
+
+        if (isExternalSection(line, section)) {
+            createExternalSection(section, line);
+        }
+
+        if (isInternalSection(line, section)) {
+            createInternalSection(section);
+        }
+    }
+
+    /* 구간 종점 밖에 구간이 추가되는 경우 */
+    private boolean isExternalSection(Line line, Section section) {
+        return section.isSameUpStationId(line.getDownStationId()) ||
+                section.isSameDownStationId(line.getUpStationId());
+    }
+
+    /* 구간 목록 내부에 구간이 추가되는 경우 */
+    private boolean isInternalSection(Line line, Section section) {
+        return section.isSameUpStationId(line.getUpStationId()) ||
+                section.isSameDownStationId(line.getDownStationId());
+    }
+
+    private void validUnknownStation(Section section) {
+        if (sectionList.stream().noneMatch(streamSection -> streamSection.isKnownSection(section))) {
+            throw new IllegalArgumentException("상행역, 하행역이 노선에 포함되어 있지 않습니다. 상행역ID:" + section.getUpStationId() +
+                    ", 하행역ID:" + section.getDownStationId());
+        }
+    }
+
+    private void createExternalSection(Section section, Line line) {
+        sectionList.add(section);
+        section.registerLine(line);
+    }
+
+    private void createInternalSection(Section section) {
+        validInternal(section);
+        List<Section> createUpdateSection = new ArrayList<>();
+
+        /* 상행역과 연결 된 역을 추가할 경우 */
+        sectionList.stream().filter(streamSection -> streamSection.isSameUpStation(section))
+                .findFirst()
+                .ifPresent(streamSection -> createUpdateSection.addAll(createUpSection(streamSection, section)));
+        /* 하행역과 연결 된 역을 추가할 경우 */
+        sectionList.stream().filter(streamSection -> streamSection.isSameDownStation(section))
+                .findFirst()
+                .ifPresent(streamSection -> createUpdateSection.addAll(createDownSection(streamSection, section)));
+
+        /* 노선 목록을 lazy 하게 변경할 목적 */
+        sectionList.addAll(createUpdateSection);
+    }
+
+    private List<Section> createUpSection(Section source, Section add) {
+        sectionList.remove(source);
+        source.updateDistance(source.minusDistance(add.getDistance()));
+        source.updateUpStationId(add.getDownStationId());
```

Comment:
source를 제거하지않고 update만 해도 되지 않을까요? 🤔
```suggestion
        source.updateBy(add);
 
```

Reply:
동일한 피드백 반영 사항으로 링크로 대체하겠습니다 ! 😆 [답변 링크](https://github.com/next-step/atdd-subway-admin/pull/971#discussion_r1030954765)